vortexor/src/rpc_load_balancer.rs

```diff
@@ -0,0 +1,218 @@
+                                Ok((_subscription, receiver)) => {
+                                    while !exit.load(Ordering::Relaxed) {
+                                        match receiver.recv_timeout(Duration::from_millis(100))
+                                        {
+                                            Ok(slot) => {
+                                                last_slot_update = Instant::now();
+
+                                                server_to_slot.1.store(slot.slot, Ordering::Relaxed);
+                                                datapoint_info!(
+                                                        "rpc_load_balancer-slot_count",
+                                                        "url" => ws_url_no_token,
+                                                        ("slot", slot.slot, i64)
+                                                );
+
+                                                {
+                                                    let old_slot = highest_slot.fetch_max(slot.slot, Ordering::Relaxed);
+                                                    if slot.slot > old_slot {
+                                                        if let Err(e) = slot_sender.send(slot.slot)
+                                                        {
+                                                            error!("error sending slot: {e}");
+                                                            break;
+                                                        }
+                                                    }
+                                                }
+                                            }
+                                            Err(RecvTimeoutError::Timeout) => {
+                                                // RPC servers occasionally stop sending slot updates and never recover.
+                                                // If enough time has passed, attempt to recover by forcing a new connection
+                                                if last_slot_update.elapsed() >= Self::DISCONNECT_WEBSOCKET_TIMEOUT
+                                                {
+                                                    datapoint_error!(
+                                                        "rpc_load_balancer-force_disconnect",
+                                                        "url" => ws_url_no_token,
+                                                        ("event", 1, i64)
+                                                    );
+                                                    break;
+                                                }
+                                            }
+                                            Err(RecvTimeoutError::Disconnected) => {
+                                                info!("slot subscribe disconnected. url: {ws_url_no_token}");
```

Comment:
should this be a warning? how often do we expect disconnects form the subscriptions?